### DISCUSSION TITLE
Spotlight2D

### DISCUSSION THREAD

**Discussion by amap1380** [godotengine/godot-proposals#4177] [2D]
It can be achieved with regular Light2D but it needs custom sprites which are inflexible and can't be changed.
With a Spotlight2D node its much easier to do custom lighting like line of sight and etc.

**Comment by eon-s** [comment]
Godot 4 has a PointLight2D node, maybe it can just get a `from` and `to` angles to restrict the region and work as a 2D spotlight.

**Comment by Calinou** [comment]
With [GradientTexture2D](https://github.com/godotengine/godot/pull/53234) and its [`3.x` backport](https://github.com/godotengine/godot/pull/54824), you can already do this using a Light2D and a GradientTexture2D. You'll need Godot 3.5beta1 or 4.0alpha for this, as GradientTexture2D is not available in Godot 3.4.x.
